### PR TITLE
Add .NET test application for functional tests

### PR DESCRIPTION
**Pull request by jvoravong**
**Description:** <Describe what has changed.>
- Adds the files needed to create a Docker image containing a .NET test application
- The functional tests that will use this application will be added in a future PR.
- Created Image: https://quay.io/repository/splunko11ytest/dotnet_test?tab=tags&tag=latest

### CODE REVIEW

File: functional_tests/testdata/dotnet/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM mcr.microsoft.com/dotnet/aspnet:7.0 AS base
```
Comment (pjanotti) [inline functional_tests/testdata/dotnet/Dockerfile:1]: Not mandatory but I would recommend you go straight to 8 - 7 reaches EOL in May, the odd numbers are not LTS. Similar at line 5.

File: functional_tests/testdata/dotnet/DotNetTestApp.csproj [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (pjanotti) [inline functional_tests/testdata/dotnet/DotNetTestApp.csproj:4]: net8.0

**Review by pjanotti — APPROVED** [review]
LGTM, I do recommend going to net8.0, but it can be done separately.

File: functional_tests/testdata/dotnet/DotNetTestApp.csproj (hunk 2 above)
Comment (jvoravong) [reply functional_tests/testdata/dotnet/DotNetTestApp.csproj]: Done

File: functional_tests/testdata/dotnet/Dockerfile (hunk 1 above)
Comment (jvoravong) [reply functional_tests/testdata/dotnet/Dockerfile]: Done

**Pull request merged**